tests/database/query.test.ts

```diff
@@ -1468,60 +1585,85 @@ describe('Query Tests', function() {
   });
 
   it('handles removing a queried element', async function() {
-    var ref = <Reference>getRandomNode();
+    var ref = (getRandomNode() as Reference);
 
     var val;
+    const ea = new EventAccumulator(1);
     ref.limitToLast(1).on('child_added', function(snap) {
       val = snap.val();
+      ea.addEvent();
     });
 
-    await ref.set({a: 1, b: 2});
+    ref.set({a: 1, b: 2});
     expect(val).to.equal(2);
 
-    return ref.child('b').remove();
+    ref.child('b').remove();
+
+    await ea.promise;
+
+    expect(val).to.equal(1);
   });
 
-  it('.startAt().limitToFirst(1) works.', async function() {
-    var ref = <Reference>getRandomNode();
+  it('.startAt().limitToFirst(1) works.', function(done) {
+    var ref = (getRandomNode() as Reference);
+    ref.set({a: 1, b: 2});
+    
     var val;
     ref.startAt().limitToFirst(1).on('child_added', function(snap) {
       val = snap.val();
+      if (val === 1) {
+        done();
+      }
     });
-    await ref.set({a: 1, b: 2});
   });
 
   it('.startAt().limitToFirst(1) and then remove first child (case 1664).', async function() {
-    var ref = <Reference>getRandomNode();
+    var ref = (getRandomNode() as Reference);
+    ref.set({a: 1, b: 2});
 
+    const ea = new EventAccumulator(1);
     var val;
     ref.startAt().limitToFirst(1).on('child_added', function(snap) {
       val = snap.val();
+      ea.addEvent();
     });
 
-    await ref.set({a: 1, b: 2});
+    await ea.promise;
+    expect(val).to.equal(1);
+
+    ea.reset();
+    ref.child('a').remove();
 
-    return ref.child('a').remove();
+    await ea.promise;
+    expect(val).to.equal(2);
   });
 
-  it('.startAt() with two arguments works properly (case 1169).', async function() {
-    var ref = <Reference>getRandomNode();
-    await ref.set({ 'Walker': { name: 'Walker', score: 20, '.priority': 20 }, 'Michael': { name: 'Michael', score: 100, '.priority': 100 } });
-    const s = await ref.startAt(20, 'Walker').limitToFirst(2).once('value');
-    var childNames = [];
-    s.forEach(function(node) { childNames.push(node.key); });
-    expect(childNames).to.deep.equal(['Walker', 'Michael']);
+  it('.startAt() with two arguments works properly (case 1169).', function(done) {
+    var ref = (getRandomNode() as Reference);
+    const data = { 
+      'Walker': { 
+        name: 'Walker', 
+        score: 20, 
+        '.priority': 20 
+      }, 
+      'Michael': { 
+        name: 'Michael', 
+        score: 100, 
+        '.priority': 100 
+      } 
+    };
+    ref.set(data, function() {
+      ref.startAt(20, 'Walker').limitToFirst(2).on('value', function(s) {
+        var childNames = [];
+        s.forEach(function(node) { childNames.push(node.key); });
+        expect(childNames).to.deep.equal(['Walker', 'Michael']);
+        done();
+      });
+    });
```

Comment:
I only refactored completely those functions that chained `runs`/`waitsFor` blocks. If there was a path were there was a smaller code delta (in this case the function waited for an assignment of a truthy value to `done`) then I favored that.

This is something we could clean up as well if we wanted and I'm totally in favor of this. Just already a lot of code change and wanted to make review easy where I could.